hack/generated/controllers/generic_controller.go

```diff
@@ -0,0 +1,531 @@
+			}
+
+			status, err = gr.getStatus(ctx, deployment.Properties.OutputResources[0].ID, data)
+			if err != nil {
+				return errors.Wrap(err, "getting status from ARM")
+			}
+		}
+
+		err = mutData.Update(deployment, status)
+		if err != nil {
+			return errors.Wrap(err, "updating metaObj")
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		// This is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+		// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
+		return ctrl.Result{}, errors.Wrap(client.IgnoreNotFound(err), "patching")
+	}
+
+	// TODO: Could somehow have a method that grouped both of these calls
+	currentState := data.GetResourceProvisioningState()
+	data.log.Info("Monitoring deployment", "action", string(action), "id", deployment.Id, "state", currentState)
+	gr.Recorder.Event(data.metaObj, v1.EventTypeNormal, string(action), fmt.Sprintf("Monitoring Azure deployment ID=%q, state=%q", deployment.Id, currentState))
+
+	// We do two patches here because if we remove the deployment before we've actually confirmed we persisted
+	// the resource ID, then we will be unable to get the resource ID the next time around. Only once we have
+	// persisted the resource ID can we safely delete the deployment
+	if deployment.IsTerminalProvisioningState() && !data.GetShouldPreserveDeployment() {
+		data.log.Info("Deleting deployment", "ID", deployment.Id)
+		err = gr.Patch(ctx, data, func(ctx context.Context, mutData *ReconcileMetadata) error {
+			err := gr.ARMClient.DeleteDeployment(ctx, deployment.Id)
+			if err != nil {
+				return errors.Wrapf(err, "deleting deployment %q", deployment.Id)
+			}
+			deployment.Id = ""
+			deployment.Name = ""
+
+			err = mutData.Update(deployment, status)
+			if err != nil {
+				return errors.Wrap(err, "updating metaObj")
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			// This is a superfluous error as per https://github.com/kubernetes-sigs/controller-runtime/issues/377
+			// The correct handling is just to ignore it and we will get an event shortly with the updated version to patch
+			return ctrl.Result{}, errors.Wrap(client.IgnoreNotFound(err), "patching")
+		}
+	}
+
+	result := ctrl.Result{}
+	// TODO: This is going to be common... need a wrapper/helper somehow?
+	if !deployment.IsTerminalProvisioningState() {
+		result = ctrl.Result{
+			RequeueAfter: 5 * time.Second,
+		}
+	}
+	return result, err
+}
+
+//////////////////////////////////////////
+// Other helpers
+//////////////////////////////////////////
+
+func (gr *GenericReconciler) constructArmResource(ctx context.Context, data *ReconcileMetadata) (genruntime.ArmResource, error) {
+	_, armResourceSpec, err := ResourceSpecToArmResourceSpec(ctx, gr.ResourceResolver, data.metaObj)
+	if err != nil {
+		return nil, errors.Wrapf(err, "converting to armResourceSpec")
+	}
+	// TODO: Not setting status here
```

Comment:
Does this mean you ARE setting status and need to not do so, or that you're not setting status and you should?

Reply:
clarified